app/core/abilities/abilities.ts

```diff
@@ -13024,6 +13026,55 @@ export class SpinOutStrategy extends AbilityStrategy {
   }
 }
 
+export class RockArtilleryStrategy extends AbilityStrategy {
+  process(
+    pokemon: PokemonEntity,
+    board: Board,
+    target: PokemonEntity,
+    crit: boolean
+  ) {
+    super.process(pokemon, board, target, crit, true)
+    const numberOfRocks = Math.round(
+      ([10, 20, 30][pokemon.stars - 1] ?? 30) * (1 + pokemon.ap / 100)
+    )
+    const damage = [10, 20, 30][pokemon.stars - 1] ?? 30
+
+    const enemies = board.cells.filter(
+      (cell) => cell && cell.team !== pokemon.team
+    ) as PokemonEntity[]
+
+    for (let i = 0; i < numberOfRocks; i++) {
+      const randomEnemy = pickRandomIn(enemies)
+      if (randomEnemy) {
+        const adjacentCells = board.getAdjacentCells(
+          randomEnemy.positionX,
+          randomEnemy.positionY,
+          true
+        )
+        const targetCell = pickRandomIn(adjacentCells)
```

Comment:
if i understood correctly, the ability would do zero damage if the target is the only enemy remaining ?